metadata-ingestion/src/datahub/ingestion/api/decorators.py

```diff
@@ -34,7 +34,9 @@ def platform_name(
 
     def wrapper(cls: Type) -> Type:
         cls.get_platform_name = lambda: platform_name
-        cls.get_platform_id = lambda: id or platform_name.lower().replace(" ", "-")
+        cls.get_platform_id = lambda *args: id or platform_name.lower().replace(
+            " ", "-"
+        )
```

Comment:
browse path source helper relies on `platform` attribute of source or `platform` attribute of source's config to generate platform instance however, the `platform` is set to `lookml` in lookml source and is used in stale metadata removal job id generation. 

`get_platform_id` added as part of `@platform_name` wrapper (used for grouping sources together) can be used as a standard mechanism for identifying `platform` of a source. It accepts separate platform name and id and can be flexibly used to separate display name from platform's id.

Reply:
Looks like this is not working for s3 where platform is picked from config.

Reply:
reverted this logic and added logic to prefer "platform_name" from config if its present.